src/buildstream/storage/directory.py

```diff
@@ -16,328 +17,499 @@
 #
 #  Authors:
 #        Jim MacArthur <jim.macarthur@codethink.co.uk>
+#        Tristan van Berkom <tristan.vanberkom@codethink.co.uk>
 
 """
-Directory
-=========
-
-This is a virtual Directory class to isolate the rest of BuildStream
-from the backing store implementation.  Sandboxes are allowed to read
-from and write to the underlying storage, but all others must use this
-Directory class to access files and directories in the sandbox.
-
-See also: :ref:`sandboxing`.
-
+Directory - Interfacing with files
+==================================
+The Directory class is given to plugins by way of the :class:`.Sandbox`
+or in some other instances, and allows plugins to interface with files
```

Comment:
Word missing after 'or'?

Reply:
I did `s/or/and/` here and improved the grammar of this phrase.